packages/discovery-react-components/src/components/SearchResults/utils/getDocumentTitle.ts

```diff
@@ -0,0 +1,18 @@
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import get from 'lodash/get';
+
+export const getDocumentTitle = (
+  resultTitleField: string,
+  result?: DiscoveryV2.QueryResult
+): string => {
+  if (result) {
+    const documentId: string | undefined = get(result, 'document_id');
+    const filename: string | undefined = get(result, 'extracted_metadata.filename');
+    const userSelectedTitle: string | undefined = get(result, resultTitleField);
+    const title: string = userSelectedTitle
+      ? userSelectedTitle
+      : get(result, 'extracted_metadata.title') || filename || documentId;
+    return title;
```

Comment:
There's a lot of unnecessary code, some duplication and types that are getting in the way and making this much more difficult to read than necessary.

```suggestion
    return result[resultTitleField] || 
        get(result, 'extracted_metadata.title') ||
        get(result, 'extracted_metadata.filename') ||
        result.document_id;
```